### ISSUE TITLE
TypeError: array_map(): Argument #2 ($array) must be of type array, null given in BaseFileUpload component

### ISSUE THREAD

**Issue by ralphjsmit** [laravel-filament/filament#1037]
### Package

filament/forms

### Package Version

2.7.1

### Laravel Version

8.75.0

### Livewire Version

2.8.2

### PHP Version

8.1.0

### Bug description

We were running unit tests on GitHub Actions and I noticed that the tests started failing due to a weird error message in Filament. The interesting thing is that the tests all work locally, whilst they're not working anymore on GitHub Actions. 

This is the error (attached below).

I double checked and I saw that it was introduced by version 2.7.1.

v2.7.0 and lower: works
2.7.1 and higher: breaks

The error is in the BaseFileUpload Component.

I'm doing nothing special in the tests. I just set the $image property to null for the tests with the $component->set(...), just for a regular LW component.

Would be great if you could check this!

### Steps to reproduce

Create a Livewire component and a Filament UploadedFile inside. Create a unit test for that component and use `$component->set('image', null)`. This breaks the test on GitHub Actions (tested on ubuntu and macos).

### Relevant log output

```shell
1) Tests\Feature\Livewire\Dashboard\Products\CreateTest::can_create_products
16
TypeError: array_map(): Argument #2 ($array) must be of type array, null given
17

18
/Users/runner/work/X/X/vendor/filament/forms/src/Components/BaseFileUpload.php:356
19
/Users/runner/work/X/X/vendor/filament/forms/src/Components/BaseFileUpload.php:61
20
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:36
21
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/Util.php:40
22
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:81
23
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:37
24
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/Container.php:653
25
/Users/runner/work/X/X/vendor/filament/forms/src/Components/Concerns/EvaluatesClosures.php:15
26
/Users/runner/work/X/X/vendor/filament/forms/src/Components/Concerns/HasState.php:70
27
/Users/runner/work/X/X/vendor/filament/forms/src/Concerns/HasState.php:49
28
/Users/runner/work/X/X/vendor/filament/forms/src/Concerns/HasState.php:69
29
/Users/runner/work/X/X/vendor/filament/forms/src/Concerns/HasState.php:200
30
/Users/runner/work/X/X/app/Http/Livewire/Dashboard/Products/Create.php:66
31
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:36
32
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/Util.php:40
33
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:93
34
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php:37
35
/Users/runner/work/X/X/vendor/livewire/livewire/src/ComponentConcerns/HandlesActions.php:149
36
/Users/runner/work/X/X/vendor/livewire/livewire/src/HydrationMiddleware/PerformActionCalls.php:38
37
/Users/runner/work/X/X/vendor/livewire/livewire/src/LifecycleManager.php:89
38
/Users/runner/work/X/X/vendor/livewire/livewire/src/Connection/ConnectionHandler.php:13
39
/Users/runner/work/X/X/vendor/livewire/livewire/src/Controllers/HttpConnectionHandler.php:20
40
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php:48
41
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/Route.php:262
42
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/Route.php:205
43
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/Router.php:695
44
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:128
45
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103
46
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/Router.php:697
47
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/Router.php:672
48
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/Router.php:636
49
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Routing/Router.php:625
50
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:167
51
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:128
52
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:103
53
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:142
54
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:111
55
/Users/runner/work/X/X/vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php:510
56
/Users/runner/work/X/X/vendor/livewire/livewire/src/Testing/TestableLivewire.php:196
57
/Users/runner/work/X/X/vendor/livewire/livewire/src/Testing/MakesHttpRequestsWrapper.php:29
58
/Users/runner/work/X/X/vendor/livewire/livewire/src/Testing/TestableLivewire.php:197
59
/Users/runner/work/X/X/vendor/livewire/livewire/src/Testing/TestableLivewire.php:181
60
/Users/runner/work/X/X/vendor/livewire/livewire/src/Testing/Concerns/MakesCallsToComponent.php:142
61
/Users/runner/work/X/X/vendor/livewire/livewire/src/Testing/Concerns/MakesCallsToComponent.php:37
62
/Users/runner/work/X/X/vendor/livewire/livewire/src/Testing/Concerns/MakesCallsToComponent.php:31
63
/Users/runner/work/X/X/tests/Feature/Livewire/Dashboard/Products/CreateTest.php:54
64
/Users/runner/work/X/X/vendor/phpunit/phpunit/phpunit:98
```

**Comment by danharrin** [comment]
Fixed by #1038.

**Comment by ralphjsmit** [comment]
Thanks Dan!

Sorry to bug you, I just noticed that the `2.7.4` version gives another error:

```
TypeError : Filament\Forms\Components\BaseFileUpload::Filament\Forms\Components\{closure}(): Argument #2 ($value) must be of type array, Livewire\TemporaryUploadedFile given
```

The first error only happened when I set the image property to `null`, the second error happens when I set the image property to an `UploadedFile`.
